### PR TITLE
support viewport selection in maya2019 VP2

### PR DESCRIPTION
**Pull request by pilarmolinalopez**
## Description 
Implements VP2 selection in maya2019 thanks to the new MPxDrawOverride::userSelect() interface

### Added

- Implementation of new ProxyDrawOverride::userSelect() provided with maya2019
- Implementation of ProxyShape::getShapeSelectionMask() 

## Checklist (Please do not remove this line)
- [x] Make sure the Title and Description of the PR make sense and  provide sufficient context for your work
- [ ] Do any added files have the correct AL Apache Licence Header?
- [x] Are there Doxygen comments in the headers?
- [ ] Are any new features, behaviour changes documented in the .md format [documentation](https://github.com/AnimalLogic/AL_USDMaya/docs)?
- [ ] Have you added, updated tests to cover new features and behaviour changes?
- [x] Have you filled out at least one changelog entry?

### CODE REVIEW

File: lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShape.h [hunk 1]
```diff
@@ -829,6 +830,10 @@ class ProxyShape
   AL_USDMAYA_PUBLIC
   SdfPathVector getPrimPathsFromCommaJoinedString(const MString &paths) const;
 
+  /// \brief  Returns the selection mask of the shape
+  AL_USDMAYA_PUBLIC
+  MSelectionMask	getShapeSelectionMask() const override;
```
Comment (robthebloke) [inline lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShape.h]: Strange tab between return type, and method name...

**Comment by pilarmolinalopez** [comment]
Thanks @robthebloke, I made those changes. Hopefully everything is good now. Sorry for the annoying formatting issues.

**Comment by fabal** [comment]
Thanks @pilarmolinalopez for this change. We were hoping to merge your work this week but the move to gcc 6 is taking some time on our side (we need to update quite a lot of dependencies). Next week seems more reasonable.
Also, just to let you know that AL_USDMaya's pull requests should normally be created against `develop` instead of `master`.

**Comment by pilarmolinalopez** [comment]
Thanks @fabal . And sorry about that, I'll make sure to create the pull request against develop next time.

**Comment by elrond79** [comment]
I think it's also worth noting that, with the changes in userSelect, there should no longer be a need for two different selection methods - that was only there because, with the old VP1 based way, there was a bit of a pick-your poison scenario.  Though I haven't tested this specific PR, I did play with the new VP2 userSelect in 2019, and I think it should resolve all the issues - ie, it should be able to play nicely with undo AND deal with maya shapes.

So moving forward, I think we should drop the fetching of AL_usdmaya_selectMode, and eliminate the "if(1 == selectionMode)" branch.

Also - I know this is from the original (VP1) code, but: we shouldn't need our own removeVariantFromPath function.  SdfPath:: StripAllVariantSelections should do the job.

**Comment by pilarmolinalopez** [comment]
Thanks @elrond79. Got it. I can try those changes and add them to this pull request before it's merged, if they work.

**Comment by pilarmolinalopez** [comment]
I replaced the `removeVariantFromPath` function with `SdfPath::StripAllVariantSelections`. 
Regarding what @elrond79  mentioned about the new `userSelect` and undo, it still doesn't seem to work in the maya2019 version I'm testing with. I'm going to investigate more. When I resolve it I think it can go into a new pull request, so this one should be ready to merge now if you guys agree.

**Comment by elrond79** [comment]
@pilarmolinalopez  - interesting, when I was testing the new selection functionality, I'd swear it worked with undo.  Of course, I was just testing with one of their devkit plugins (rockingTransform), so maybe there's something particular about the AL code that's causing problems.  I'll see if I can take a look.

In the meantime, I did notice one other issue - in your implementation of getShapeSelectionMask, you're returning kSelectMeshes.  However, the AL plugin provides it's own selection mask, specifically for proxy shapes, which you should use - ProxyShape::s_selectionMaskName.  (Ideally, I would suggest  caching the creation of the selection mask from the name, and then returning a copy of that.  Because the api for getShapeSelectionMask returns an MSelectionMask, and not a const MSelectionMask&, the copy is unavoidable, but at least we can avoid having to do the name-lookup every time getShapeSelectionMask is called!)

**Comment by pilarmolinalopez** [comment]
@elrond79, it was just a very quick test of the "undo" issue on my side. I'd like to test it further because I may have missed something. I'll let you know when I have updates.
Regarding the selection mask, I remember seeing that mask but I didn't use it because it makes impossible to select Maya objects that are in front of USD prims. I just tested it and confirmed that's the case. If I click on a Maya object in front of a USD prim it will select the USD prim.

**Comment by fabal** [comment]
Superseded by #118

**Pull request closed without merging**